src/databases/functions/users/user-set-karma-score.js

```diff
@@ -1,10 +1,34 @@
 const {User} = require('../../models');
+const {QueryTypes} = require('sequelize');
+const db = require('../../models/');
 
 module.exports = async (userId, combined_user_score, karma_score) => {
   if (!userId) return null;
   try {
+    let total_follower = 0;
+    const queryTotalFollower = `
+      SELECT user_id_followed, COUNT(user_id_follower) AS follower_count
+      FROM user_follow_user
+      WHERE
+      is_anonymous = false
+      AND user_id_followed = :userId
+      GROUP BY user_id_followed
+    `;
+
+    const resultTotalFollower = await db.sequelize.query(queryTotalFollower, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      },
+      raw: true
+    });
+
+    if (queryTotalFollower.length > 0) {
+      total_follower = resultTotalFollower[0]?.follower_count;
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There seems to be a logical error in your code. You are checking the length of `queryTotalFollower` which is a string, not the result of the query (`resultTotalFollower`). This will always evaluate to true if `queryTotalFollower` is not an empty string, regardless of whether any followers were found or not.
```diff
-    if (queryTotalFollower.length > 0) {
+    if (resultTotalFollower.length > 0) {
       total_follower = resultTotalFollower[0]?.follower_count;
     }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->